torax/_src/imas_tools/output/core_profiles.py

```diff
@@ -0,0 +1,482 @@
+  ids.vacuum_toroidal_field.r0 = geometry[0].R_major
+  ids.vacuum_toroidal_field.b0 = [
+      geometry_slice.B_0 for geometry_slice in geometry
+  ]  # TODO: Check sign(s) once TORAX COCOS will be defined.
+  ip = []
+  v_loop = []
+  I_bs = []
+  li_3 = []
+  beta_pol = []
+  beta_tor = []
+  beta_tor_norm = []
+  t_e_volume_average = []
+  n_e_volume_average = []
+  for cp_slice in core_profiles:
+    ip.append(-1 * cp_slice.Ip_profile_face[-1])
+    v_loop.append(cp_slice.v_loop_lcfs)
+  for ppo_slice in post_processed_outputs:
+    I_bs.append(-1 * ppo_slice.I_bootstrap)
+    li_3.append(ppo_slice.li3)
+    beta_pol.append(ppo_slice.beta_pol)
+    beta_tor.append(ppo_slice.beta_tor)
+    beta_tor_norm.append(ppo_slice.beta_N)
+    t_e_volume_average.append(ppo_slice.T_e_volume_avg * 1e3)
+    n_e_volume_average.append(ppo_slice.n_e_volume_avg)
+  ids.global_quantities.ip = ip
+  ids.global_quantities.current_bootstrap = I_bs
+  ids.global_quantities.v_loop = v_loop
+  ids.global_quantities.li_3 = li_3
+  ids.global_quantities.beta_pol = beta_pol
+  ids.global_quantities.beta_tor = beta_tor
+  ids.global_quantities.beta_tor_norm = beta_tor_norm
+  # Temperatures are converted from keV to eV(IMAS standard unit).
+  ids.global_quantities.t_e_volume_average = t_e_volume_average
+  ids.global_quantities.n_e_volume_average = n_e_volume_average
+  ids.global_quantities.ion_time_slice = times[0]
+  main_ion = runtime_params_provider(
+      times[0]
+  ).plasma_composition.main_ion.fractions
+  impurities = post_processed_outputs[0].impurity_species.keys()
+  num_ions = len(main_ion) + len(impurities)
+  ids.global_quantities.ion.resize(num_ions)
+  for ion in range(num_ions):
+    ids.global_quantities.ion[ion].t_i_volume_average.resize(len(times))
+    ids.global_quantities.ion[ion].n_i_volume_average.resize(len(times))
+
+
+def _fill_profiles_1d(
+    ids: ids_toplevel.IDSToplevel,
+    runtime_params_provider: build_runtime_params.RuntimeParamsProvider,
+    post_processed_outputs: list[post_processing.PostProcessedOutputs],
+    core_profiles: list[state.CoreProfiles],
+    core_sources: list[source_profiles.SourceProfiles],
+    geometry: list[geometry.Geometry],
+    times: array_typing.FloatVector,
+) -> None:
+  ids.profiles_1d.resize(len(times))
+  for i in range(len(times)):
+    t = times[i]
+    geometry_slice = geometry[i]
+    cp_state = core_profiles[i]
+    cs_state = core_sources[i]
+    post_processed_outputs_slice = post_processed_outputs[i]
+    runtime_params = runtime_params_provider(t)
+
+    ids.profiles_1d[i].time = t
+    _fill_grid(ids, i, geometry_slice, cp_state)
+    _fill_ions(ids, i, runtime_params, post_processed_outputs_slice, cp_state)
+    _fill_currents(ids, i, post_processed_outputs_slice, cp_state, cs_state)
+    ids.profiles_1d[i].electrons.temperature = (
+        cp_state.T_e.cell_plus_boundaries() * 1e3
+    )
+    ids.profiles_1d[i].electrons.density = cp_state.n_e.cell_plus_boundaries()
```

Comment:
just a general comment here for all the `cell_plus_boundaries` quantities. Can we extract them just once and then populate them using that rather than recomputing each time.